loader/standard_loader.py

```
@@ -74,65 +139,72 @@ def _load_bundle(self, bundle_uuid, metadata_dict, data_objects):
-            file_version = self._get_file_version(file_info)
-            cloud_urls = self._get_cloud_urls(file_info)
+        for data_file in data_files:
+            filename, file_uuid, cloud_urls, bundle_uuid, file_guid, file_version, = data_file
             logger.debug(f'Attempting to upload data file: {filename} with uuid:version {file_uuid}:{file_version}...')
-            file_uuid, file_version, filename = \
+            file_uuid, file_version, filename, already_present = \
                 self.dss_uploader.upload_cloud_file_by_reference(filename,
                                                                  file_uuid,
                                                                  cloud_urls,
                                                                  bundle_uuid,
                                                                  file_guid,
                                                                  file_version=file_version)
+            if already_present:
+                logger.info('File already present. No upload necessary.')
             logger.debug(f'...Successfully uploaded data file: {filename} with uuid:version {file_uuid}:{file_version}')
             file_info_list.append(dict(uuid=file_uuid, version=file_version, name=filename, indexed=False))
 
         # load bundle
         self.dss_uploader.load_bundle(file_info_list, bundle_uuid)
 
+    def _parse_all_bundles(self, input_json):
+        """Parses all raw json bundles"""
+        if type(input_json) is not list:
+            raise ParseError(f"Json file is misformatted. Expected type: list, actually type {type(input_json)}")
+
+        for count, bundle in enumerate(input_json):
+            try:
+                parsed_bundle = self._parse_bundle(bundle)
+                self.bundles_parsed.append(parsed_bundle)
+            except ParseError:
+                logger.exception(f'Could not parse bundle {count + 1}')
```

Because you know it is a `ParseError` and the `ParseErrors` have messages, I would log the message here, instead of explicitly logging the error before throwing the `ParseError`.

---

This comment has not been addressed yet.

---

Sorry I should have responded before. This functionality is actually already there. See https://docs.python.org/3/library/logging.html#logging.exception